File: Neetcode/LongestRepeatCharReplace.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = dict()
        l = 0
        r = 0
        maxfreq = 0
        maxlen = 0
        while r < len(s):
            if s[r] in count:
                count[s[r]] += 1
            else:
                count[s[r]] = 1
            maxfreq = max(maxfreq, count[s[r]])
            if r-l+1 - maxfreq <= k:
                maxlen = max(maxlen, r-l+1)
            else:
                while r-l+1 - maxfreq > k:
                    count[s[l]] -= 1
                    l += 1
            r += 1
        return maxlen

File: Neetcode/LongestRepeatCharReplace.cpp
#include <unordered_map>
using std::unordered_map;
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> hashmap;
        int left = 0;
        int right = 0;
        int len = 0;
        while (right < s.length()){
            hashmap[s[right]] += 1;
            int maxfreq = 0;
            for (auto i: hashmap){
                maxfreq = max(maxfreq, i.second);
            }
            if (right - left + 1 - maxfreq <= k){
                len = max(right - left + 1, len);
            }
            else{
                while (right - left + 1 - maxfreq > k){
                    hashmap[s[left]] -= 1;
                    left += 1;
                }
            }
            right += 1;
        }
        return len;
    }
};
